infra/modules/providers/azure/app-gateway/main.tf

```diff
@@ -0,0 +1,72 @@
+data "azurerm_resource_group" "appgateway" {
+  name      = "${var.resource_group_name}"
+}
+
+data "azurerm_virtual_network" "appgateway" {
+    name                = "${var.virtual_network_name}"
+    resource_group_name = "${data.azurerm_resource_group.appgateway.name}"
+}
+data "azurerm_subnet" "appgateway" {
+    name                    = "${var.subnet_name}"
+    resource_group_name     = "${data.azurerm_resource_group.appgateway.name}"
+    virtual_network_name    = "${data.azurerm_virtual_network.appgateway.name}"
+}
+
+data "azurerm_public_ip" "appgateway" {
+    name                    = "${var.public_ip_name}"
+    resource_group_name     = "${data.azurerm_resource_group.appgateway.name}"
+}
+
+resource "azurerm_application_gateway" "appgateway" {
+  name                = "${var.appgateway_name}"
+  resource_group_name = "${data.azurerm_resource_group.appgateway.name}"
+  location            = "${data.azurerm_resource_group.appgateway.location}"
+  tags                = "${var.resource_tags}"
+
+  sku {
+    name     = "${var.appgateway_sku_name}"
+    tier     = "${var.appgateway_tier}"
+    capacity = "${var.appgateway_capacity}"
+  }
+
+  gateway_ip_configuration {
+    name      = "${var.appgateway_ipconfig_name}"
+    subnet_id = "${data.azurerm_subnet.appgateway.id}"
+  }
+
+  frontend_port {
+    name = "${var.appgateway_frontend_port_name}"
+    port = 80
+  }
+
+  frontend_ip_configuration {
+    name                  = "${var.appgateway_frontend_ip_configuration_name}"
+    public_ip_address_id  = "${data.azurerm_public_ip.appgateway.id}"
+  }
+
+  backend_address_pool {
+    name = "${var.appgateway_backend_address_pool_name}"
+  }
+
+  backend_http_settings {
+    name                  = "${var.appgateway_backend_http_setting_name}"
+    cookie_based_affinity = "${var.backend_http_cookie_based_affinity}"
+    port                  = 80
```

Comment:
Can you add an optional variable for the backend port to avoid the port 80 hard coding?